chain/network/src/peer_manager/network_state/mod.rs

```diff
@@ -303,4 +352,64 @@ impl NetworkState {
             .with_span_context(),
         );
     }
+
+    /// Check for edges indicating that
+    /// a) there is a peer we should be connected to, but we aren't
+    /// b) there is an edge indicating that we should be disconnected from a peer, but we are connected.
+    /// Try to resolve the inconsistency.
+    pub fn fix_local_edges(self: &Arc<Self>, clock: &time::Clock) {
+        let local_edges = self.routing_table_view.get_local_edges();
+        let tier2 = self.tier2.load();
+        for edge in local_edges {
+            let node_id = self.config.node_id();
+            let other_peer = edge.other(&node_id).unwrap();
+            match (tier2.ready.get(other_peer), edge.edge_type()) {
```

Comment:
So, it seems like just commenting out the whole `match` doesn't cause any tests to fail. Is there a way we can test this? Or are we too far away from being able to speed the time up?

Reply:
I've added a test, but left some cases to cover as a todo.